conan/api/subapi/cache.py

```diff
@@ -131,50 +131,49 @@ def save(self, package_list, tgz_path, no_source=False):
-                    folder = os.path.relpath(pkg_folder, cache_folder)
-                    folder = folder.replace("\\", "/")  # make win paths portable
-                    pref_bundle["package_folder"] = folder
-                    out.info(f"Saving {pref}: {folder}")
-                    tgz.add(os.path.join(cache_folder, folder), folder, recursive=True)
-                    if os.path.exists(pref_layout.metadata()):
-                        metadata_folder = os.path.relpath(pref_layout.metadata(), cache_folder)
-                        metadata_folder = metadata_folder.replace("\\", "/")  # make paths portable
-                        pref_bundle["metadata_folder"] = metadata_folder
-                        out.info(f"Saving {pref} metadata: {metadata_folder}")
-                        tgz.add(os.path.join(cache_folder, metadata_folder), metadata_folder,
-                                recursive=True)
-            serialized = json.dumps(package_list.serialize(), indent=2)
-            info = tarfile.TarInfo(name="pkglist.json")
-            data = serialized.encode('utf-8')
-            info.size = len(data)
-            tgz.addfile(tarinfo=info, fileobj=BytesIO(data))
-            tgz.close()
+                    tar_files[f"{recipe_folder}/{f}"] = path
+            path = os.path.join(cache_folder, recipe_folder, DOWNLOAD_EXPORT_FOLDER, METADATA)
+            if os.path.exists(path):
+                tar_files[f"{recipe_folder}/{DOWNLOAD_EXPORT_FOLDER}/{METADATA}"] = path
+
+            for pref, pref_bundle in package_list.prefs(ref, ref_bundle).items():
+                pref_layout = cache.pkg_layout(pref)
+                pkg_folder = pref_layout.package()
+                folder = os.path.relpath(pkg_folder, cache_folder)
+                folder = folder.replace("\\", "/")  # make win paths portable
+                pref_bundle["package_folder"] = folder
+                out.info(f"Saving {pref}: {folder}")
+                tar_files[folder] = os.path.join(cache_folder, folder)
+
+                if os.path.exists(pref_layout.metadata()):
+                    metadata_folder = os.path.relpath(pref_layout.metadata(), cache_folder)
+                    metadata_folder = metadata_folder.replace("\\", "/")  # make paths portable
+                    pref_bundle["metadata_folder"] = metadata_folder
+                    out.info(f"Saving {pref} metadata: {metadata_folder}")
+                    tar_files[metadata_folder] = os.path.join(cache_folder, metadata_folder)
+
+        # Create a temporary file in order to reuse compress_files functionality
+        serialized = json.dumps(package_list.serialize(), indent=2)
+        pkglist_path = os.path.join(tempfile.gettempdir(), "pkglist.json")
+        save(pkglist_path, serialized)
+        tar_files["pkglist.json"] = pkglist_path
+        compress_files(tar_files, os.path.basename(tgz_path), os.path.dirname(tgz_path), compresslevel, recursive=True)
+        remove(pkglist_path)
```

Comment:
If we want to keep this remove, shouldn't it be affecting the whole folder

Reply:
What do you mean? I am not sure I understood your comment